src/distilabel/pipeline/local.py

```diff
@@ -285,41 +294,122 @@ def signal_handler(signumber: int, frame: Any) -> None:
 
 
 class _WriteBuffer:
+    """Class in charge of sending the batched contents to a buffer and writing
+    those to files under a given folder.
+
+    As batches are received, they are added to the buffer and once each buffer
+    is full, the content is written to a parquet file.
+    """
+
+    _type_map: Dict[type, pa.DataType] = {
+        int: pa.int64(),
+        float: pa.float64(),
+        str: pa.string(),
+    }
+
     def __init__(self, path: "PathLike", leaf_steps: Set[str]) -> None:
-        path = Path(path)
-        if path.suffix == "":
-            path = path / "data.jsonl"
-        self._path = path
+        """
+        Args:
+            path (PathLike): Folder where the files will be written, the idea
+                is for this path to be in the cache folder under /data.
+            leaf_steps (Set[str]): Leaf steps from the dag of the pipeline.
```

Comment:
We agreed on removing the types out of the docstrings as those will be automatically retrieved from the function signature 👍🏻

Reply:
```suggestion
            path: Folder where the files will be written, the idea
                is for this path to be in the cache folder under /data.
            leaf_steps: Leaf steps from either the DAG of the Pipeline.
```